PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/Util.java

```diff
@@ -578,4 +640,189 @@ public static String generateInstreamUriForGam(String adUnit, int w, int h, Map<
         return uri;
     }
 
+    private static final String GAM_VIEW_CLASS = "com.google.android.gms.ads.doubleclick.PublisherAdView";
+    private static final String GAM_CUSTOM_TEMPLATE_AD_CLASS = "com.google.android.gms.ads.formats.NativeCustomTemplateAd";
+    private static final String MOPUB_NATIVE_AD_CLASS = "com.mopub.nativeads.NativeAd";
+    private static final String INNER_HTML_SCRIPT = "document.body.innerHTML";
+
+    public static void findNative(@NonNull Object object, @NonNull PrebidNativeAdListener listener) {
+        String objectClassName = object.getClass().getCanonicalName();
+        if (GAM_VIEW_CLASS.equals(objectClassName)) {
+            View adView = (View) object;
+            findNativeInGAMPublisherAdView(adView, listener);
+        } else if (MOPUB_NATIVE_AD_CLASS.equals(objectClassName)) {
+            findNativeInMoPubNativeAd(object, listener);
+        } else if (implementsInterface(object, GAM_CUSTOM_TEMPLATE_AD_CLASS)) {
+            finaNativeInGAMCustomTemplateAd(object, listener);
+        } else {
+            listener.onPrebidNativeNotFound();
+        }
+    }
+
+    private static boolean implementsInterface(@NonNull Object object, @NonNull String it) {
+        for (Class c : object.getClass().getInterfaces()) {
+            Log.d("Prebid", c.getCanonicalName());
```

Comment:
Use LogUtil to log messages.
Same for other necessary logs too.

Reply:
Done.